sdk/storage/Azure.Storage.DataMovement.Blobs/api/Azure.Storage.DataMovement.Blobs.net6.0.cs

```diff
@@ -54,10 +54,13 @@ public partial class BlobStorageResourceContainerOptions
     public partial class BlobStorageResourceOptions
     {
         public BlobStorageResourceOptions() { }
-        public Azure.Storage.Blobs.Models.AccessTier? AccessTier { get { throw null; } set { } }
-        public Azure.Storage.Blobs.Models.BlobHttpHeaders HttpHeaders { get { throw null; } set { } }
-        public System.Collections.Generic.IDictionary<string, string> Metadata { get { throw null; } set { } }
-        public System.Collections.Generic.IDictionary<string, string> Tags { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<Azure.Storage.Blobs.Models.AccessTier?> AccessTier { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<string> CacheControl { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<string> ContentDisposition { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<string> ContentEncoding { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<string> ContentLanguage { get { throw null; } set { } }
+        public Azure.Storage.DataMovement.DataTransferProperty<string> ContentType { get { throw null; } set { } }
```

Comment:
We talked about this offline. I think we prefer to leave these broken out, which will allow us more flexibility in the feature to allow certain options to be set, without making major API changes.

Reply:
Two follow-up here:
- What happens if someone chooses to preserve some and not preserve others during single-call copy? Since there is only the one setting for all properties. This also applies to metadata. Combining all HTTP headers into one could at least make this less likely to happen but it could still happen with metadata and headers.
- I think we discussed adding `ContentHash` back in? I think we should do that if these stay as seperate settings.

Reply:
- If someone decides to do that. Then we have to set the copy-properties header to false, and then set the properties that they did want to preserve manually. Basically we would be doing the same thing we would be doing on a chunked copy (aka preserving the properties by manually setting each property value). Same for metadata.
- If we combine it into one, it removes the flexibility in the future when we do get the ask to preserve some of the properties but not all. Which is the flexibility they have today in the base storage SDK.
- I will add `ContentHash` back in, but for future reference `ContentHash` is one of those properties that easily evolve into a feature ask to take many forms given our past interactions with customers regarding this property.

Reply:
I did the above with the exception of `ContentHash`. This will be done later on which we talked about offline.